sklearn/metrics/tests/test_classification.py

```diff
@@ -724,9 +725,30 @@ def test_cohen_kappa():
     )
 
 
-def test_matthews_corrcoef_nan():
-    assert matthews_corrcoef([0], [1]) == 0.0
-    assert matthews_corrcoef([0, 0], [0, 1]) == 0.0
+@pytest.mark.parametrize(
+    "zero_division",
+    ["warn", 0, 1, np.nan],
+)
+def test_matthews_corrcoef_nan(zero_division):
```

Comment:
Could we make a common test for all metrics that support this feature.

I would write two parameterized tests: (i) specifically for warn and (ii) another for the other.